src/Illuminate/Database/Eloquent/SoftDeletingScope.php

```diff
@@ -128,4 +144,17 @@ protected function addOnlyTrashed(Builder $builder)
             return $builder;
         });
     }
+
+    /**
+     * Add the with-trashed extension to the builder.
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder  $builder
+     * @return void
+     */
+    protected function addOnlyTrashedIf(Builder $builder)
+    {
+        $builder->macro('onlyTrashedIf', function (Builder $builder, $condition) {
+            return $condition ? $builder->onlyTrashed() : $builder->withoutTrashed();
```

Comment:
I'm not sure this is right. `withoutTrashed` is not the opposite of `onlyTrashed`. 
What happens if someone calls `Model::withTrashedIf(true)->onlyTrashedIf(false)`?

Reply:
Why is it not an opposite?

|                     | withoutTrashed | onlyTrashed |
|---------------------|:--------------:|:-----------:|
| Trashed records     |        ❌       |      ✔️      |
| Not trashed records |        ✔️       |      ❌      |

Reply:
Default case is trashed not visible, not-trashed visible.
`withTrashed` forces trashed to visible, doesn't touch not-trashed.
`withoutTrashed` forces trashed to invisible, doesn't touch not-trashed.
`onlyTrashed` forces not-trashed to invisble, doesn't touch trashed.

`! onlyTrashed` would jus

Reply:
Semantically, saying "I don't want only the trashed records" is not the same as saying "I only want the not-trashed records".

The issue here would be chaining logic. If I say "I want all the records, even the trashed ones, but optionally only the trashed ones", and the latter condition applies `withoutTrashed` when false, I end up getting "All the records, even the trashed ones, but without the trashed ones".

Reply:
> Passing false to these functions should simply not apply the filter, rather than applying whatever the opposite filter is.

Yeah, I kinda agree with you on that.
I'll wait for Taylor to express his opinion though, because pushing yet another change would make things yet more complicated.